mautic/mautic
# Fixing campaign contact field condition on state field

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | \
| Related developer documentation PR URL | \
| Issue(s) addressed                     | Fixes https://github.com/mautic/mautic/issues/10868



#### Description:

It's not possible to create a campaign contact field condition with the state field because the state field is not a flat array but multidimensional array so it cannot be `array_flip`ed. This PR is flattening it when checking the values before save.

#### Steps to test this PR:


0. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
1. Create a new campaign, and open the editor, select a segment from source
2. Select a condition, field value.
3. Select the value from state, and select any state.
4. Try to update the node, and the errors come.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10961"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
